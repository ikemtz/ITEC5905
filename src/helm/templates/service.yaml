apiVersion: v1
kind: Service
metadata:
  name: {{ include "itec5905.fullname" . }}
  labels:
    {{- include "itec5905.labels" . | nindent 4 }}
spec:
  type: {{ .Values.service.type }}
  ports:
    - port: {{ .Values.service.port }}
      targetPort: http
      protocol: TCP
      name: http
  selector:
    {{- include "itec5905.selectorLabels" . | nindent 4 }}
