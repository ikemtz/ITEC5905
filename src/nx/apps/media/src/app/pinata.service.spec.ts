import { Test } from '@nestjs/testing';
import { faker } from '@faker-js/faker'
import { PinataService } from './pinata.service';
import { readFileSync } from 'fs';

describe('PinataService', () => {
  let service: PinataService;

  beforeAll(async () => {
    const app = await Test.createTestingModule({
      providers: [PinataService],
    }).compile();

    service = app.get<PinataService>(PinataService);
  });

  it('should get a file"', async () => {
    const result = await service.getFiles('');
    expect(result).toMatchSnapshot();
  });

  it('should upload a file"', async () => {
    const fileContents = btoa(readFileSync('./swagger.json', { encoding: 'utf-8' }));
    const resp = await service.uploadFile({
      fileName: `${faker.person.firstName()}.json`,
      referenceId: faker.commerce.productName(),
      referenceName: faker.commerce.productDescription(),
      content: fileContents,
    });
    expect(resp).toMatchSnapshot();
  });
});
