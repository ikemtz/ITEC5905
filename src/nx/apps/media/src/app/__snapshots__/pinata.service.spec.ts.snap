// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinataService should get a file" 1`] = `
[
  {
    "date_pinned": "2023-06-01T21:42:26.942Z",
    "date_unpinned": "2023-06-02T02:05:02.252Z",
    "id": "6afa92e1-3275-4731-8330-a0225188e1ea",
    "ipfs_pin_hash": "Qmb5ZrrEbtj3chk3u7S8VdkJRpqyiePFvarUjUcdQ9buye",
    "metadata": {
      "keyvalues": {},
      "name": "swagger.json",
    },
    "regions": [
      {
        "currentReplicationCount": 1,
        "desiredReplicationCount": 1,
        "regionId": "EU-CENTRAL-1",
      },
    ],
    "size": 1622,
    "user_id": "0529d2ad-98a8-4cb4-b6a7-aae2641e9b21",
  },
]
`;

exports[`PinataService should upload a file" 1`] = `
{
  "IpfsHash": "Qmb5ZrrEbtj3chk3u7S8VdkJRpqyiePFvarUjUcdQ9buye",
  "PinSize": 1622,
  "Timestamp": "2023-06-02T03:14:28.717Z",
}
`;
